WAIUA/Commands/Main.cs

```diff
@@ -250,189 +272,223 @@ public static void GetLatestVersion()
+                var response = client.Put(request);
+                string content = client.Execute(request).Content;
 
-                    content = content.Replace("[", "");
-                    content = content.Replace("]", "");
+                content = content.Replace("[", "");
+                content = content.Replace("]", "");
 
-                    var uinfo = JsonConvert.DeserializeObject(content);
-                    JToken uinfoObj = JObject.FromObject(uinfo);
-                    gameName = uinfoObj["GameName"].Value<string>();
-                    gameTag = uinfoObj["TagLine"].Value<string>();
-                    IGN = gameName + "#" + gameTag;
-                }
+                var uinfo = JsonConvert.DeserializeObject(content);
+                JToken uinfoObj = JObject.FromObject(uinfo);
+                gameName = uinfoObj["GameName"].Value<string>();
+                gameTag = uinfoObj["TagLine"].Value<string>();
+                IGN = gameName + "#" + gameTag;
             }
-            catch (Exception e)
+            catch (Exception)
             {
-                System.Diagnostics.Debug.WriteLine(e);
+                IGN = null;
             }
             return IGN;
         }
 
-        public static Boolean LiveMatchID(CookieContainer jar)
+        private static bool LiveMatchID(CookieContainer jar)
         {
             try
             {
                 string url = $"https://glz-{Shard}-1.{Region}.a.pvp.net/core-game/v1/players/{PPUUID}";
-                RestClient client = new RestClient(url);
-                client.CookieContainer = jar;
-                RestRequest request = new RestRequest(Method.GET);
+                RestClient client = new(url)
+                {
+                    CookieContainer = jar
+                };
+                RestRequest request = new(Method.GET);
                 request.AddHeader("X-Riot-Entitlements-JWT", $"{EntitlementToken}");
                 request.AddHeader("Authorization", $"Bearer {AccessToken}");
                 string response = client.Execute(request).Content;
-                //string response = DoCachedRequest(Method.GET, $"https://glz-{Shard}-1.{Region}.a.pvp.net/core-game/v1/players/{PPUUID}", true, jar, true);
+                //string response = DoCachedRequest(Method.GET, $"https://glz-{Shard}-1.{Region}.a.pvp.net/core-game/v1/players/{PUUID}", true, jar, true);
                 var matchinfo = JsonConvert.DeserializeObject(response);
                 JToken matchinfoObj = JObject.FromObject(matchinfo);
                 Matchid = matchinfoObj["MatchID"].Value<string>();
                 return true;
             }
-            catch (Exception e)
+            catch (Exception)
             {
-                System.Diagnostics.Debug.WriteLine(e);
                 return false;
             }
         }
 
-        public bool LiveMatchSetup()
+        public bool LiveMatchChecks()
```

Comment:
@NNebus Can you check if this flow works or if there are any edge cases that could make it fail? The only thing I can think of right now is if Someone uses it with one account and then switches account and tried again which will make the match detection fail as it is checking the original account.

Reply:
> @NNebus Can you check if this flow works or if there are any edge cases that could make it fail? The only thing I can think of right now is if Someone uses it with one account and then switches account and tried again which will make the match detection fail as it is checking the original account.

Hey @Soneliem I noticed this behavior as well. 
Imo this issue occurs when someone hits the button to check the match but is still in loading screen to start the match. In this time a player might _think_ he's already in a match but the api returns no running match. 
Since this request takes a few second, following request might return an existing running match and everything runs as expected.